pychromecast/dial.py

```diff
@@ -71,9 +71,17 @@ def _get_host_from_zc_service_info(service_info: zeroconf.ServiceInfo):
     return (host, port)
 
 
-def _get_status(services, zconf, path, secure, timeout, context):
+def _get_status(services, zconf, path, timeout, context, secure=None):
     """Query a cast device via http(s)."""
 
+    if secure is None:
+        try:
+            return _get_status(services, zconf, path, timeout / 2, context, secure=True)
+        except (urllib.error.HTTPError, urllib.error.URLError):
+            return _get_status(
+                services, zconf, path, timeout / 2, context, secure=False
+            )
+
```

Comment:
This logic should not be here, please move it to `get_device_info`

Reply:
Moved